### PR TITLE
Make a list of Synapse IDs public

### PR DESCRIPTION
**Pull request by philerooski**

### CODE REVIEW

**Comment by kdaily** [comment]
Like the idea of first part (changing perms) - should generalize it more to set specific permissions in a user friendly way.

Second part (move to a public folder) already exists (at least in command line with `synapse mv`).

File: pythonSynapseUtils/makepublic.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    syn.login()
+    if parentId:
+        print("Moving Files to public Folder...")
+        for i in ids:
+            args = _synObj(i, parentId)
+            move(args, syn)
+    else:
+        print("Making public...")
+        for i in ids:
+            syn.setPermissions(i, accessType=['READ'], warn_if_inherits=False)
```
Comment (kkdang) [inline pythonSynapseUtils/makepublic.py]: Phil, could you add a short test script for this code? In particular I am wondering here if the principalID will default to Public or not? It wouldn't hurt to be explicit in the call.

**Comment by philerooski** [comment]
I like the idea of generalizing the permissions "switch" to different permissions. And I did steal the move functionality from the CLI tool. Considering that this will likely be used as part of a bash script, I'm in favor of removing the move functionality from my code, unless @kkdang has anything against it.

I'll push a commit once I get that and the test code worked out.

**Comment by philerooski** [comment]
I generalized the code to be able to add or remove arbitrary (valid) permissions and added tests. Just call nosetests to run.

**Pull request merged**